app/models/search_facet.rb

```diff
@@ -358,12 +358,14 @@ def generate_bq_query_string
   def generate_array_query
     "SELECT DISTINCT id, name FROM(SELECT id_col AS id, name_col as name " + \
     "FROM #{CellMetadatum::BIGQUERY_TABLE}, UNNEST(#{self.big_query_id_column}) AS id_col WITH OFFSET id_pos, " + \
-    "UNNEST(#{self.big_query_name_column}) as name_col WITH OFFSET name_pos WHERE id_pos = name_pos)"
+    "UNNEST(#{self.big_query_name_column}) as name_col WITH OFFSET name_pos WHERE id_pos = name_pos) WHERE id IS NOT NULL " + \
```

Comment:
I'm unclear how we could have a situation where `id` is `NULL` - is this because these new "custom" facets aren't pairs, and are not required?

Reply:
yeah, it's possible that it's just relics from wonky populates in my development db, but I had null values that were throwing off the results.  It might not be a real risk for production, but I figure it can't hurt